src/terminal/display.c

```diff
@@ -246,7 +246,8 @@ int __guac_terminal_set(guac_terminal_display* display, int row, int col, int co
 
 guac_terminal_display* guac_terminal_display_alloc(guac_client* client,
         const char* font_name, int font_size, int dpi,
-        guac_terminal_color* foreground, guac_terminal_color* background) {
+        guac_terminal_color* foreground, guac_terminal_color* background,
+        const guac_terminal_color (*palette)[256]) {
```

Comment:
I left this the way it is, because foreground/background are currently part of the character attributes, so I wasn't sure if I should pull them out into a new struct with the palette.